File: tests/transpiler/x/php/tree_sum.php
<?php
function sum_tree($t) {
  return match($t) {
    $Leaf => 0,
    Node($left, $value, $right) => sum_tree($left) + $value + sum_tree($right),
};
}
$t = ["left" => $Leaf, "value" => 1, "right" => ["left" => $Leaf, "value" => 2, "right" => $Leaf]];
echo rtrim((is_float(sum_tree($t)) ? sprintf("%.15f", sum_tree($t)) : sum_tree($t))), PHP_EOL;
?>

File: archived/tests/transpiler/x/fs/tree_sum.fs
// Generated 2025-07-21 21:44 +0700

type Tree =
    | Leaf
    | Node of obj * int * obj
let rec sum_tree t =
    match t with
| Leaf -> 0
| Node left value right -> ((sum_tree left) + value) + (sum_tree right)
let t: Tree = Node(Leaf, 1, Node(Leaf, 2, Leaf))
printfn "%s" (string (sum_tree t))
